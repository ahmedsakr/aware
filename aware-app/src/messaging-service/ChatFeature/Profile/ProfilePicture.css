.profile-picture-canvas {
    display: inline-block;
    width: 24px;
    height: 24px;
    margin-right: 10px;
}

.profile-picture-activity-online {
    opacity: 1.0;
}

.profile-picture-activity-offline {
    opacity: 0.4;
}

.profile-picture-canvas > img {
    width: 100%;
    height: 100%;
    border-radius: 50px;
}