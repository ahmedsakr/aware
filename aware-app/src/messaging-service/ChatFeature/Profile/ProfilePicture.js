import React, { Component } from 'react';
import './ProfilePicture.scss'

class ProfilePicture extends Component {
    render() {
        const instance = "profile-picture-" + this.props.instance + "-size";

        return(
            <div class={"profile-picture-canvas " + instance + " profile-picture-activity-" + this.props.activity}>
                <img alt={this.props.picture} src={process.env.PUBLIC_URL + this.props.picture}/>
            </div>
        );
    }
}

export default ProfilePicture;
