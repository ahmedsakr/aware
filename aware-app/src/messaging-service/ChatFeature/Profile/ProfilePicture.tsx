import React from 'react';
import './ProfilePicture.scss'

type ProfilePictureProps = {
    picture: string,
    activity: string,
    instance: string
};

type ProfilePictureState = { };

export default class ProfilePicture extends React.Component<ProfilePictureProps, ProfilePictureState> {
    render(): JSX.Element {
        const instance = "profile-picture-" + this.props.instance + "-size";

        return(
            <div className={"profile-picture-canvas " + instance + " profile-picture-activity-" + this.props.activity}>
                <img alt={this.props.picture} src={process.env.PUBLIC_URL + this.props.picture}/>
            </div>
        );
    }
}
