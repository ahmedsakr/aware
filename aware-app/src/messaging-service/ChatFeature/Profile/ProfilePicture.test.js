import React from 'react';
import { shallow } from 'enzyme';

import ProfilePicture from './ProfilePicture';

describe('ProfilePicture', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<ProfilePicture debug />);
  
    expect(component).toMatchSnapshot();
  });
});
