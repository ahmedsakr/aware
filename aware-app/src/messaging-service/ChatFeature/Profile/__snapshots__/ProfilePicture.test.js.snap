version https://git-lfs.github.com/spec/v1
oid sha256:d15d6afe39b3146d62104fd30aeabb00bcf87b2eaae3e0cb601dc0bf20f7216c
size 264
